subprojects/docs/src/docs/release/notes.md

```diff
@@ -87,7 +87,144 @@ TODO - Java lambdas are supported, and unsupported captured values are reported.
+
+Gradle [Kotlin DSL scripts](userguide/kotlin_dsl.html#sec:scripts) are compiled by Gradle during the configuration phase of your build.
+
+Deprecation warnings found by the Kotlin compiler are now reported on the console.
+This makes it easier to spot usages of deprecated members in your build scripts.
+
+```text
+> Configure project :
+w: build.gradle.kts:4:5: 'getter for uploadTaskName: String!' is deprecated. Deprecated in Java
+```
+
+Moreover, Kotlin DSL script compilation errors are now always reported in the file order
+This makes it easier to figure out the first root cause of a script compilation failure.
+
+```text
+* Where:
+Build file 'build.gradle.kts' line: 5
+
+* What went wrong:
+Script compilation errors:
+
+  Line 5: functionDoesNotExist()
+          ^ Unresolved reference: functionDoesNotExist
+
+  Line 8: doesNotExistEither = 23
+          ^ Unresolved reference: doesNotExistEither
+
+2 errors
+```
+
+#### `kotlin-dsl` plugin improvements
+
+The [Kotlin DSL Plugin](userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin) provides a convenient way to develop Kotlin-based projects that contribute build logic.
+
+##### Easier customization of Kotlin options
+
+Thanks to the Kotlin Gradle Plugin now using Gradle lazy properties, the `kotlin-dsl` plugin does not use `afterEvaluate {}` for configuring Kotlin compiler options anymore.
+This allows for easier customization of Kotlin options in your build logic without requiring `afterEvaluate {}`.
+
+```kotlin
+plugins {
+    `kotlin-dsl`
+}
+tasks.withType<KotlinCompile>().configureEach {
+    compilerOptions {
+        languageVersion.set(KotlinVersion.KOTLIN_1_9)
+    }
+}
+```
+
+The standalone script compilation is also now configured to skip the pre-release check in order to allow referencing Kotlin code compiled with more recent Kotlin language versions on a best-effort basis.
+
+##### Published with proper licensing information
+
+The `kotlin-dsl` plugin is now published to the Gradle Plugin Portal with proper licensing information in its metadata.
+The plugin is published under the same license as the Gradle Build Tool: the Apache License  Version 2.0.
+
+This makes it easier to use the `kotlin-dsl` plugin in an enterprise setting where published licensing information is required. 
+
+#### Precompiled Kotlin script plugins improvements
+
+In addition to plugins written as standalone projects, Gradle also allows you to provide build logic written in Kotlin as [precompiled script plugins](userguide/custom_plugins.html#sec:precompiled_plugins).
+You write these as `*.gradle.kts` files in `src/main/kotlin` directory.
+
+##### Respect `--offline`
+
+Building precompiled script plugins now respects the [--offline](userguide/command_line_interface.html#sec:command_line_execution_options) command line execution option.
+
+This makes it easier to use Gradle plugins that react to `--offline` from precompiled script plugins.
+
+##### Less verbose compilation
```

Comment:
💭 
Is it sad that I kind of want to be needlessly verbose in this section? 😄

Reply:
:) I added some details. Better now?